### ISSUE TITLE
Change CellSizeCache interface for the better perfomance

### ISSUE THREAD

**Issue by arusakov** [bvaughn/react-virtualized#486]
Currently `CellSizeCache` interface has both `has` and `get` methods. But `CellMeasurer` call `get` method after `has` method always:
- [getColumnWidth](https://github.com/bvaughn/react-virtualized/blob/2a721a74a1cd48bcfda7b2495578fa517d01239c/source/CellMeasurer/CellMeasurer.js#L74-L76)
- [getRowHeight](https://github.com/bvaughn/react-virtualized/blob/2a721a74a1cd48bcfda7b2495578fa517d01239c/source/CellMeasurer/CellMeasurer.js#L98-L100)

It leads to double hashtable lookup inside default `CellSizeCache` (and in any other cache implementations I think).

I think we can remove `has` methods and change `get` to something like this:
```js
getRowHeight(index: number): number | null | undefined;
getColumnWidth(index: number): number | null | undefined;
```
`Undefined` or `null` (simple check with == null) result will be interpreted like no value, any `number` (including zero) will be interpreted like a value.

**Comment by bvaughn** [comment]
This is a good suggestion. Would you be interested in contributing a PR that implements this?

**Comment by arusakov** [comment]
@bvaughn Yes, I am.

**Comment by bvaughn** [comment]
Fantastic! It's all yours then.

**Comment by bvaughn** [comment]
Closed via PR #487

**Comment by bvaughn** [comment]
This change was just released as version 8.6.1. I gave you a special shout-out in the [CHANGELOG](https://github.com/bvaughn/react-virtualized/blob/master/CHANGELOG.md#861).

**Comment by arusakov** [comment]
@bvaughn Thank you. I was glad to help.
